### ISSUE TITLE
Instructions in README.md do not mention dependence on cylc-ui

### ISSUE THREAD

**Issue by jhaiduce** [cylc/cylc-uiserver#164]
**Describe the bug**
The instructions in README.md indicate that cylc-uiserver can be installed with pip and invoked with jupyter. From experimentation and reading error messages and documentation I found that 

- jupyterhub needs to be run from the build directory, or otherwise told to run cylc-uiserver
- cylc-ui needs to be built and cylc-uiserver's STATIC argument needs to point to the cylc-ui dist directory

However, neither of these is stated in README.md. Perhaps this is not yet fully documented because the procedure is expected to change in the near future, but in the meantime it would be easier for new users to have documentation that reflects the current state of the code.

**Release version(s) and/or repository branch(es) affected?**
current master

**Steps to reproduce the bug**

Following the instructions from README.md:
1. Install cylc-uiserver: `pip install cylc-uiserver` or `pip install .`
2. Run `jupyterhub`
3. Visit jupyterhub server in a browser, log in, and click "Launch server"

**Expected behavior**
From the text in README.md, the above commands should run the Cylc web UI.

**Console output**

If jupyterhub is run from somewhere other than the cylc-uiserver checkout directory (and the python `notebook` package has not been installed):

```
[I 2020-11-09 17:59:31.981 JupyterHub log:174] 302 GET /user/jhaiduce/api/sessions?_=1604962736446 -> /hub/user/jhaiduce/api/sessions?_=1604962736446 (@::1) 0.99ms
[I 2020-11-09 17:59:31.983 JupyterHub log:174] 302 GET /user/jhaiduce/api/terminals?_=1604962736447 -> /hub/user/jhaiduce/api/terminals?_=1604962736447 (@::1) 0.60ms
[W 2020-11-09 17:59:31.997 JupyterHub base:1261] Failing suspected API request to not-running server: /hub/user/jhaiduce/api/sessions
[E 2020-11-09 17:59:31.997 JupyterHub log:174] 503 GET /hub/user/jhaiduce/api/sessions?_=1604962736446 (jhaiduce@::1) 5.96ms
[W 2020-11-09 17:59:31.999 JupyterHub base:1261] Failing suspected API request to not-running server: /hub/user/jhaiduce/api/terminals
[E 2020-11-09 17:59:32.000 JupyterHub log:174] 503 GET /hub/user/jhaiduce/api/terminals?_=1604962736447 (jhaiduce@::1) 8.15ms
[I 2020-11-09 17:59:32.755 JupyterHub log:174] 302 GET /user/jhaiduce/tree -> /hub/user/jhaiduce/tree (@::1) 0.81ms
[E 2020-11-09 17:59:32.805 JupyterHub log:174] 503 GET /hub/user/jhaiduce/tree (jhaiduce@::1) 29.60ms
[I 2020-11-09 17:59:34.632 JupyterHub spawner:1417] Spawning jupyterhub-singleuser --port=57184
Failed to set groups [Errno 1] Operation not permitted
Traceback (most recent call last):
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/singleuser.py", line 24, in <module>
    import notebook
ModuleNotFoundError: No module named 'notebook'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jhaiduce/Library/Python/3.8/bin/jupyterhub-singleuser", line 5, in <module>
    from jupyterhub.singleuser import main
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/singleuser.py", line 26, in <module>
    raise ImportError("JupyterHub single-user server requires notebook >= 4.0")
ImportError: JupyterHub single-user server requires notebook >= 4.0
[I 2020-11-09 17:59:35.609 JupyterHub log:174] 302 GET /hub/spawn/jhaiduce?next=%2Fhub%2Fuser%2Fjhaiduce%2Ftree -> /hub/spawn-pending/jhaiduce?next=%2Fhub%2Fuser%2Fjhaiduce%2Ftree (jhaiduce@::1) 1007.16ms
[I 2020-11-09 17:59:35.628 JupyterHub pages:347] jhaiduce is pending spawn
[I 2020-11-09 17:59:35.632 JupyterHub log:174] 200 GET /hub/spawn-pending/jhaiduce?next=%2Fhub%2Fuser%2Fjhaiduce%2Ftree (jhaiduce@::1) 6.67ms
ERROR:asyncio:Task exception was never retrieved
future: <Task finished name='Task-31' coro=<BaseHandler.spawn_single_user() done, defined at /Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/handlers/base.py:741> exception=HTTPError()>
Traceback (most recent call last):
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/handlers/base.py", line 938, in spawn_single_user
    await gen.with_timeout(
tornado.util.TimeoutError: Timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/handlers/base.py", line 969, in spawn_single_user
    raise web.HTTPError(
tornado.web.HTTPError: HTTP 500: Internal Server Error (Spawner failed to start [status=1]. The logs for jhaiduce may contain details.)
[W 2020-11-09 18:00:03.695 JupyterHub user:687] jhaiduce's server never showed up at http://127.0.0.1:57184/user/jhaiduce/ after 30 seconds. Giving up
[E 2020-11-09 18:00:03.714 JupyterHub gen:598] Exception in Future <Task finished name='Task-32' coro=<BaseHandler.spawn_single_user.<locals>.finish_user_spawn() done, defined at /Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/handlers/base.py:845> exception=TimeoutError("Server at http://127.0.0.1:57184/user/jhaiduce/ didn't respond in 30 seconds")> after timeout
    Traceback (most recent call last):
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/gen.py", line 593, in error_callback
        future.result()
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/handlers/base.py", line 852, in finish_user_spawn
        await spawn_future
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/user.py", line 668, in spawn
        await self._wait_up(spawner)
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/user.py", line 715, in _wait_up
        raise e
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/user.py", line 682, in _wait_up
        resp = await server.wait_up(
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/utils.py", line 229, in wait_for_http_server
        re = await exponential_backoff(
      File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/jupyterhub/utils.py", line 177, in exponential_backoff
        raise TimeoutError(fail_message)
    TimeoutError: Server at http://127.0.0.1:57184/user/jhaiduce/ didn't respond in 30 seconds
    
[I 2020-11-09 18:00:03.737 JupyterHub log:174] 200 GET /hub/api/users/jhaiduce/server/progress (jhaiduce@::1) 27875.38ms
```

Assuming jupyterhub is run from the cylc-uiserver checkout directory:

```
[I 2020-11-09T17:35:21.166 JupyterHub log:174] 302 GET /user/jhaiduce/subscriptions -> /hub/user/jhaiduce/subscriptions (@::1) 0.88ms
[I 2020-11-09T17:35:22.382 JupyterHub log:174] 302 GET /user/jhaiduce/ -> /hub/user/jhaiduce/ (@::1) 0.77ms
[W 2020-11-09T17:35:22.404 JupyterHub base:392] Invalid or expired cookie token
[I 2020-11-09T17:35:22.405 JupyterHub log:174] 302 GET /hub/user/jhaiduce/ -> /hub/login?next=%2Fhub%2Fuser%2Fjhaiduce%2F (@::1) 1.31ms
[I 2020-11-09T17:35:22.458 JupyterHub log:174] 200 GET /hub/login?next=%2Fhub%2Fuser%2Fjhaiduce%2F (@::1) 31.12ms
[W 2020-11-09T17:35:22.574 JupyterHub log:174] 404 GET /hub/logo (@::1) 0.73ms
[I 2020-11-09T17:35:27.682 JupyterHub base:707] User logged in: jhaiduce
[I 2020-11-09T17:35:27.683 JupyterHub log:174] 302 POST /hub/login?next=%2Fhub%2Fuser%2Fjhaiduce%2F -> /hub/user/jhaiduce/ (jhaiduce@::1) 245.70ms
[E 2020-11-09T17:35:27.704 JupyterHub log:174] 503 GET /hub/user/jhaiduce/ (jhaiduce@::1) 9.59ms
[W 2020-11-09T17:35:27.780 JupyterHub log:174] 404 GET /hub/logo (@::1) 1.00ms
[W 2020-11-09T17:35:27.819 JupyterHub log:174] 404 GET /hub/logo (@::1) 1.53ms
[I 2020-11-09T17:35:29.446 JupyterHub spawner:1417] Spawning cylc-uiserver --port=56701 -s /Users/jhaiduce/cylc-uiserver/../cylc-ui/dist
Failed to set groups [Errno 1] Operation not permitted
2020-11-09 17:35:30,152 cylc.uiserver.main INFO     JupyterHub Service Prefix: /user/jhaiduce/
2020-11-09 17:35:30,153 cylc.uiserver.main INFO     Listening on 56701 and serving static content from /Users/jhaiduce/cylc-uiserver/../cylc-ui/dist
2020-11-09 17:35:30,153 cylc.uiserver.main INFO     Starting Cylc UI
2020-11-09 17:35:30,153 cylc.uiserver.main INFO     /Users/jhaiduce/cylc-uiserver/../cylc-ui/dist
2020-11-09 17:35:30,167 cylc.uiserver.data_store_mgr DEBUG    register_workflow(jhaiduce|cylctest)
2020-11-09 17:35:30,167 cylc.uiserver.data_store_mgr DEBUG    register_workflow(jhaiduce|graph-introduction)
2020-11-09 17:35:30,167 cylc.uiserver.data_store_mgr DEBUG    register_workflow(jhaiduce|integer-cycling)
2020-11-09 17:35:30,168 cylc.uiserver.data_store_mgr DEBUG    register_workflow(jhaiduce|runtime-introduction)
[I 2020-11-09T17:35:30.230 JupyterHub base:854] User jhaiduce took 0.809 seconds to start
[I 2020-11-09T17:35:30.231 JupyterHub proxy:258] Adding user jhaiduce to proxy /user/jhaiduce/ => http://127.0.0.1:56701
17:35:30.233 [ConfigProxy] info: Adding route /user/jhaiduce -> http://127.0.0.1:56701
17:35:30.233 [ConfigProxy] info: Route added /user/jhaiduce -> http://127.0.0.1:56701
17:35:30.233 [ConfigProxy] info: 201 POST /api/routes/user/jhaiduce 
[I 2020-11-09T17:35:30.234 JupyterHub log:174] 302 GET /hub/spawn/jhaiduce?next=%2Fhub%2Fuser%2Fjhaiduce%2F -> /hub/spawn-pending/jhaiduce?next=%2Fhub%2Fuser%2Fjhaiduce%2F (jhaiduce@::1) 815.43ms
[I 2020-11-09T17:35:30.256 JupyterHub log:174] 302 GET /hub/spawn-pending/jhaiduce?next=%2Fhub%2Fuser%2Fjhaiduce%2F -> /hub/user/jhaiduce/ (jhaiduce@::1) 3.22ms
[I 2020-11-09T17:35:30.275 JupyterHub log:174] 302 GET /hub/user/jhaiduce/ -> /user/jhaiduce/?redirects=1 (jhaiduce@::1) 2.40ms
[I 2020-11-09T17:35:30.320 JupyterHub log:174] 302 GET /hub/api/oauth2/authorize?client_id=jupyterhub-user-jhaiduce&redirect_uri=%2Fuser%2Fjhaiduce%2Foauth_callback&response_type=code&state=[secret] -> /user/jhaiduce/oauth_callback?code=[secret]&state=[secret] (jhaiduce@::1) 11.35ms
[I 2020-11-09T17:35:30.362 JupyterHub log:174] 200 POST /hub/api/oauth2/token (jhaiduce@127.0.0.1) 19.92ms
[I 2020-11-09T17:35:30.374 JupyterHub log:174] 200 GET /hub/api/authorizations/token/[secret] (jhaiduce@127.0.0.1) 9.63ms
2020-11-09 17:35:30,391 tornado.application ERROR    Uncaught exception GET /user/jhaiduce/?redirects=1 (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:8000', method='GET', uri='/user/jhaiduce/?redirects=1', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/web.py", line 1701, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/web.py", line 1912, in wrapper
    return method(self, *args, **kwargs)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/web.py", line 3178, in wrapper
    return method(self, *args, **kwargs)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/cylc/uiserver/handlers.py", line 72, in get
    self.render(index, python_base_url=base_url)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/web.py", line 856, in render
    html = self.render_string(template_name, **kwargs)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/web.py", line 1002, in render_string
    t = loader.load(template_name)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/template.py", line 445, in load
    self.templates[name] = self._create_template(name)
  File "/Users/jhaiduce/Library/Python/3.8/lib/python/site-packages/tornado/template.py", line 476, in _create_template
    with open(path, "rb") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jhaiduce/cylc-uiserver/../cylc-ui/dist/index.html'
2020-11-09 17:35:30,394 tornado.access ERROR    500 GET /user/jhaiduce/?redirects=1 (127.0.0.1) 3.59ms
```

**Additional context**
I was able to get cylc-uiserver to run using the following steps:

1. `git clone https://github.com/cylc/cylc-uiserver`
2. `cd cylc-uiserver`
3. `pip install .`
4. `cd ..`
5. `git clone https://github.com/cylc/cylc-ui`
6. `cd cylc-ui`
7. `yarn install`
8. `yarn build`
9. `cd ../cylc-uiserver`
10. `jupyterhub`

**Comment by kinow** [comment]
@jhaiduce you are correct that we haven't yet finished to update the documentation. And well done on the detective work!

You can also join the chat in [our matrix room](https://matrix.to/#/#cylc-general:matrix.org) or [discourse](https://cylc.discourse.group/), where we can try to help you with issues with Cylc 7 or Cylc 8 (due to NZ/UK timezones, it's very likely there's always someone in that room).

There is some docs that were not published, but may help you too, e.g. https://github.com/cylc/cylc-admin/blob/master/docs/howto/cylc-ui.md

I am not sure how much we want to add in `cylc-uiserver` `README`, or if we want to add a link to the page generated by `cylc-doc` repository/code. Probably the latter I suspect.

**Comment by jhaiduce** [comment]
Thanks @kinow. I'm happy to help with refining the documentation too.

**Comment by oliver-sanders** [comment]
Closed by #188 - now the UI build comes packaged with the UIS.
